intro-to-functions/invoking-functions.md

```diff
@@ -0,0 +1,298 @@
+
+## Every Function Has a Responsibility
+
+Functions do things, and have a responsibility to do things when they are _invoked_.
+
+It will be easiest for us to use functions in the future when we understand what the _responsibility_ of each function is when we use it. We should describe what each function is supposed to be doing, and understand what "goes into" a function, gets computed inside of a function, and "goes out" of a function before we use it.
+
+Python has defined a lot of functions for us to use, already. Let's look at some Python functions and their responsibilities.
+
+| Name of function | Responsibility |
+| --- | --- |
+| `len` | Based on the one given value, will calculate the length of the given value. The given value is typically either a string or list.
+| `print` | Based on one given value, will display the given value to the "standard output" (like the console, terminal, bash, etc) as best it can.
+| `randint` | Based on two given values, generate a random integer in the range between the first given value and the second given value
+| `time.time` | Calculates the number of seconds passed since [epoch. For Unix system, January 1, 1970, 00:00:00 at UTC is epoch (the point where time begins).](https://en.wikipedia.org/wiki/Unix_time)
+
+Callout: Technically, `time.time` is a function named `time` inside a module `time`. Also, `random.randint` is a function named `randint` inside of a module named `random`. We won't worry about these details yet! What this means for us today is we will sometimes need these two lines of code at the top: `import random` and/or `import time`
+
+Callout: Going forward, because technically `time` and `randint` are the names of the functions, and `time.time` and `random.randint` are how we must invoke them, we will notate it as such.
+
+## Invoke Functions with `function_name()`
+
+If we know a function's name, we know its responsibility, and we know we want to _invoke_ this function on a certain line of code, we use this syntax:
+
+```python
+function_name()
+```
+
+... where `function_name` is replaced with the name of the function. We attach `()` to the right side of the function name.
+
+Let's look at those Python functions and the syntax for how to invoke them.
+
+| Name of function | Syntax to Invoke (With Zero Arguments) |
+| --- | --- |
+| `len` | `len()`
+| `print` | `print()`
+| `randint` (inside the `random` module) | `random.randint()`
+| `time` (inside the `time` module) | `time.time()`
+
+### Try It
+
+Run this code, and see what it outputs.
+
+```python
+import time
+time.time
+```
+
+Now, run this code, and observe the difference.
+
+```python
+import time
+time.time()
+```
+
+`time.time()` (with `()`) invokes the function, and gives us the number of seconds since epoch, fulfilling its responsibility. Without `()`, we are simply just acknowledging that `time.time` is a defined function.
+
```

Comment:
I'll look into this 💀

Reply:
RIP I was hoping that Learn would have a REPL RIP RIP RIP 💀 💀 💀 💀 

I'll think more on this now, but I like your suggestion about conforming these examples (and all repl-esque examples) into "Run this code in a REPL" instructions (rather than just "run this code")

Reply:
AHHHHH That's so frustrating about Learn!!